deploy/docker/notebook/base/base.yaml

```diff
@@ -421,26 +425,110 @@ addfiles:
           if "NB_UMASK" in os.environ:
               os.umask(int(os.environ["NB_UMASK"], 8))
 
-          def is_inside_path(path_to_test, path):
-              return os.path.samefile(os.path.commonpath([path, path_to_test]), path)
-        
-          # Configure file post-save hook
-          def script_post_save(model, os_path, contents_manager, **kwargs):
-              work_path = '/home/jovyan/work/'
-              shared_path = '/home/jovyan/shared/notebooks'
-
-              if not (is_inside_path(os_path, work_path) or is_inside_path(os_path, shared_path)):
-                    return
-
-              logging.getLogger().info(f"Saved notebook at {os_path}")
-              try:
-                    r = requests.post("https://polus-notebooks-hub-backend.ci.aws.labshare.org/file-data/new-file", json={"user": os.getenv("JUPYTERHUB_USER"), "filePath": os_path}, timeout=1)
-                    if r.status_code == 200:
-                        logging.getLogger().info(f"Registered {os_path} with Notebooks Hub")
-                    else:
-                        logging.getLogger().error(f"Could not register {os_path} with Notebooks Hub. Return code: {r.status_code}")
-              except Exception as e:
-                    logging.getLogger().error(f"Could not register {os_path} with Notebooks Hub", exc_info=e)
-              
+          # Configure file post-save, post-rename, and post-delete hooks
           
-          c.FileContentsManager.post_save_hook = script_post_save
\ No newline at end of file
+					#### Hook Class ####
+
+					class HookFileContentsManager(FileContentsManager):
+							"""Local spawner that runs single-user servers as the same user as the Hub itself.
+							Overrides user-specific env setup with no-ops.
+							"""
+							
+							post_delete_hook = Any(None, config=True,
```

Comment:
Could you bring the hook methods' definitions from outside of the class to the inside of it (`script_post_delete`→`post_delete_hook`, `script_post_save `→`post_save_hook` and `script_post_rename `→`post_rename_hook`).